### ISSUE TITLE
Better Errors not working with Rack 2.0.7

### ISSUE THREAD

**Issue by csalvato** [BetterErrors/better_errors#439]
Thanks for maintaining this awesome gem!  We use it daily!

Now, our problem:

We needed to update our `sidekiq` gem recently, and to do that we had to update to rack 2.0.7.   We noticed that this caused `better_errors` to stop working:

![image](https://user-images.githubusercontent.com/1158152/56439037-0931e680-62a2-11e9-9589-e6a7a417bc8e.png)

We isolated this issue to upgrading Rack 2.0.7.  If we do not update to Rack 2.0.7, better_error works fine. As soon as we update to that version (whether or not we update any other gems), it stops working.

**Comment by csalvato** [comment]
To reproduce, you can create a new Rails project and install `better_errors` via the Gemfile to see that `better_errors` will not work for a routing error to, say, `localhost:3000/ldkjgalgjk`.


Here's a fresh repo with `better_errors` installed just in case that helps: 

https://github.com/csalvato/better_errors_example

**Comment by csalvato** [comment]
I thought this might be an issue with Rack 2.0.7 directly and created a fresh Rails repo to test. At first I thought better errors wasn't working...but it turns out it is...

https://github.com/csalvato/better_errors_example

We must have a clash with another gem.

**Comment by RobinDaugherty** [comment]
It sounds like the Rails repo you built works correctly? Could you provide a copy of the broken application's Gemfile? Anything we might use to narrow it down? It's also _possible_ that it's caused by the order of the gems in the Gemfile. (I haven't seen such a problem yet, but it's theoretically possible.)

**Comment by csalvato** [comment]
Thanks for your reply @RobinDaugherty!

> Could you provide a copy of the broken application's Gemfile? 

We're hesitant to share the Gemfile publicly, but I've sent it over to the email address listed in your profile.

> Anything we might use to narrow it down?

In a fresh Rails repo I copied our `Gemfile` and `Gemfile.lock` files exactly (and created a skeleton config for any gems that required it).  The gems in both bundles are exactly the same. 

`better_errors` is working fine in the fresh Rails install, but not in our repo.

I'm thinking we may have some initializers/config that is possibly causing a problem. I don't have a great idea on where to start here aside from going through and disabling these files one by one and seeing if it gets fixed in our app's repo.  

If you have any advice on how to best debug this, please let me know!  We love `better_errors` and want to keep using it.

**Comment by csalvato** [comment]
We just spent the better part of today debugging this. I stripped our entire app down to a single controller that raises an error, and we were able to isolate the problem down to our infrastructure configuration.

We found issue #270 in this repo, and applied the fix as stated here by @augnustin:

https://github.com/BetterErrors/better_errors/issues/270#issuecomment-263125304

I'm still a bit confused as to why this was working fine in our repo until we updated to rack 2.0.7, since the issue is several years old 🤔 

But at least our issue is resolved now!
